aiohttp/http_parser.py

```diff
@@ -50,6 +49,18 @@
     HAS_BROTLI = False
 
 
+class Decompressor(Protocol):
```

Comment:
I'd still like to see this replaced with inheritance.
Might also be nice to move the inline BrotliDecoder to the same module (renaming from zlib_utils to compress or similar).

Reply:
should it be a part of the same PR?

Reply:
Yeah, still feels like it's part of refactoring the compressing logic.

Reply:
done